### PR TITLE
Support Service Principal for wal-g backup to Azure

### LINKED ISSUE #775: Support SPN for wal-g backups to Azure Blob Storage
Currently, `wal-g` authenticates to Azure Blob Storage with a Storage Account key (`AZURE_STORAGE_ACCESS_KEY`). However, `wal-g` also supports [authentication with a Service Principal](https://github.com/wal-g/wal-g/blob/master/docs/STORAGES.md#azure), which is, in general, more desirable because it allows more fine-grained access control. 

Please add passing the environment variables `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, and `AZURE_TENANT_ID` to `wal-g`. For clarity, it might be an option to pass either `AZURE_STORAGE_ACCESS_KEY` or the SPN variables.

### PR DESCRIPTION
**Pull request by stephan2012**
Signed-off-by: Stephan Austermühle <au@hcsd.de>

Closes #775

### CODE REVIEW

File: postgres-appliance/scripts/configure_spilo.py [hunk 1]
```diff
@@ -854,7 +856,31 @@ def write_wale_environment(placeholders, prefix, overwrite):
...
+        if wale.get('AZURE_CLIENT_ID') and wale.get('AZURE_CLIENT_SECRET') and wale.get('AZURE_TENANT_ID'):
+            if auth_opts == 0:
+                azure_auth.extend(['AZURE_CLIENT_ID', 'AZURE_CLIENT_SECRET', 'AZURE_TENANT_ID'])
+            auth_opts += 1
+
+        if auth_opts > 1:
+            logging.warning('Multiple authentication options configured for wal-g backup to Azure, using {}. Provide '
+                            'either AZURE_STORAGE_ACCESS_KEY or AZURE_STORAGE_SAS_TOKEN or Service Principal '
+                            '(AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_TENANT_ID) for authentication (or use '
+                            'MSI).'.format('/'.join(azure_auth)))
```
Comment (CyberDem0n) [inline postgres-appliance/scripts/configure_spilo.py:880]: With `logging` we prefer to use `%s`-like placeholders for formatting.

**Comment by CyberDem0n** [comment]
:+1:

**Comment by hughcapet** [comment]
👍

**Comment by hughcapet** [comment]
Merged. Thank you @stephan2012!

**Pull request merged**